packages/flutter_dynamic_forms_components/lib/src/components/slider/slider.yaml

```diff
@@ -0,0 +1,26 @@
+type: slider
+parentType: formElement
+properties:
+  value:
+    type: double
+    isMutable: false
```

Comment:
This should be true.
The `isMutable` means that if you set it to true, you are able to call the `changeValue` method on this property, otherwise, the value couldn't be changed.

I have also noticed that parser code is not up to date with the YAML file. The isMutable property is converted into negated isImmutable property on the parser.  So you will probably need to run
`flutter packages pub run build_runner build --delete-conflicting-outputs` again after you change this.

Note: If the example project won't compile after the generator command, just delete .dart_tool folder and reload the project. I still didn't find the source of this bug. Another problem you may hit is that it will regenerate all the components with different line endings. This command `git config --global core.autocrlf input` and running generator again fixed it for me.

Reply:
I know this is remove old ones but when run this command
`"flutter packages pub run build_runner build --delete-conflicting-outputs"`

it is removed 
      id: parserNode.getPlainStringValue("id"),
      parent: parserNode.getParentValue(parent),
      isVisible: parserNode.getIsVisible(),

values other components.

Reply:
Hm, that's strange, it looks like some new bug. It is working on my environment. Can you please submit a new issue with the `flutter doctor -v` output?

Basically, when each YAML file is processed it finds all the related YAML files. In your case when you declare `parentType: formElement`, it is scanning for `form_element.yaml` and generates all parent properties. It looks like the file is not found in your case, which is weird.

If we don't resolve it soon I can still merge it and fix the slider.yaml by myself, so we are not blocked by this issue.

Reply:
> Hm, that's strange, it looks like some new bug. It is working on my environment. Can you please submit a new issue with the `flutter doctor -v` output?
> 
> Basically, when each YAML file is processed it finds all the related YAML files. In your case when you declare `parentType: formElement`, it is scanning for `form_element.yaml` and generates all parent properties. It looks like the file is not found in your case, which is weird.
> 
> If we don't resolve it soon I can still merge it and fix the slider.yaml by myself, so we are not blocked by this issue.

Yeah, I saw that problem myself. Isn't there a command to create a specific folder instead of rebuilding all components?

Reply:
It is using the build_runner package https://pub.dev/packages/build_runner which serves as a code generator for many Flutter packages and it doesn't have a command to generate something only from the subset of the project. I would need to update the package so that you will be able to somehow specify it in the build.yaml file, but I don't think it makes much sense because the output should be still the same for all files.

I would be able to better assist you if you submit it as a separate issue with the `flutter doctor -v` output so I can see your OS and Flutter version and minimal reproducible example (for example if the issue is also in the current master branch of the components package).

Reply:
Ok. Here is the details

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.6, on Mac OS X 10.15 19A602, locale
    en-TR)
    • Flutter version 1.9.1+hotfix.6 at /Users/yasinilhan/flutter
    • Framework revision 68587a0916 (7 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/yasinilhan/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.1, Build version 11A1027
    • CocoaPods version 1.8.3

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 40.2.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.39.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.6.0

[✓] Connected device (1 available)
    • Ebru • e111cda2bd51b99d8d73cdd4a20cee33a52ec3ba • ios • iOS 13.1.2

```

Reply:
Thanks! It is working on my windows machine, but I was able to reproduce this behavior on my Mac. I will try to fix it and let you know when it is done...

Reply:
Ok, the issue is fixed and published in the pub.

You just need to call `flutter packages upgrade` so the new fixed version of the generator will be used and hopefully all models and parsers will be generated without a problem.

Reply:
Ok thanks, will test it .

Reply:
> This should be true.
> The `isMutable` means that if you set it to true, you are able to call the `changeValue` method on this property, otherwise, the value couldn't be changed.
> 
> I have also noticed that parser code is not up to date with the YAML file. The isMutable property is converted into negated isImmutable property on the parser. So you will probably need to run
> `flutter packages pub run build_runner build --delete-conflicting-outputs` again after you change this.
> 
> Note: If the example project won't compile after the generator command, just delete .dart_tool folder and reload the project. I still didn't find the source of this bug. Another problem you may hit is that it will regenerate all the components with different line endings. This command `git config --global core.autocrlf input` and running generator again fixed it for me.

Are you sure that :) about isMutable means that if i set ture i can able to call changeValue method ? Because i saw when i change it false the value changed otherwise i got this error.
`type 'ImmutableElementValue<double>' is not a subtype of type 'MutableElementValue<dynamic>' in type cast`

Reply:
yes, I just cloned your repo, changed value in your slider so the value looks like this:
```
  value:
    type: double
    isMutable: true
```
Then I called `flutter packages pub run build_runner build --delete-conflicting-outputs`

and the parser code for value looks like this:
```
value: parserNode.getValue<double>(
        "value",
        (s) => double.parse(s),
        () => 0,
        isImmutable: false,
```
So it is unchanged. And it was working for me.

It is a little bit confusing that yaml files are using isMutable and parsers are using isImmutable so the values are opposite, maybe this should be changed in the future so it is more consistent.

Reply:
Ok fixed